### PR TITLE
resource/aws_rds_cluster: Fix an enabled_cloudwatch_logs_exports type conversion error

### PR DESCRIPTION
**Pull request by shuheiktgw**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
* resource/aws_rds_cluster: Fix an enabled_cloudwatch_logs_exports type conversion error
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSRDSCluster_PointInTimeRestore'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSRDSCluster_PointInTimeRestore -timeout 120m
=== RUN   TestAccAWSRDSCluster_PointInTimeRestore
=== PAUSE TestAccAWSRDSCluster_PointInTimeRestore
=== CONT  TestAccAWSRDSCluster_PointInTimeRestore
--- PASS: TestAccAWSRDSCluster_PointInTimeRestore (456.39s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	458.258s
```

When I set `restore_to_point_in_time` block with `enabled_cloudwatch_logs_exports`, terraform-aws-provider crashes. It seems `enabled_cloudwatch_logs_exports` needs to be casted to `*schema.Set` not `[]interface{}` so I fixed it.

**Terraform configuration**

```
resource "aws_rds_cluster" "xxx" {
  cluster_identifier = "xxx"
  engine             = "aurora-postgresql"
  engine_version     = "11.7"

  db_subnet_group_name   = aws_db_subnet_group.xxx.name
  vpc_security_group_ids = [aws_security_group.xxx.id]

  db_cluster_parameter_group_name = "xxx"
  database_name                   = "xxx"
  master_username                 = "xxx"
  master_password                 = "xxx"

  enabled_cloudwatch_logs_exports = ["postgresql"]

  restore_to_point_in_time {
    source_cluster_identifier  = "xxx"
    restore_type               = "copy-on-write"
    use_latest_restorable_time = true
  }
}
```

**Crash logs**

```
panic: interface conversion: interface {} is *schema.Set, not []interface {}
2020-11-26T01:14:10.048Z [DEBUG] plugin.terraform-provider-aws_v3.18.0_x5: 
2020-11-26T01:14:10.048Z [DEBUG] plugin.terraform-provider-aws_v3.18.0_x5: goroutine 70 [running]:
2020-11-26T01:14:10.048Z [DEBUG] plugin.terraform-provider-aws_v3.18.0_x5: github.com/terraform-providers/terraform-provider-aws/aws.resourceAwsRDSClusterCreate(0xc001e29b80, 0x55ba9a0, 0xc000fb7b80, 0x0, 0xffffffffffffffff)
2020-11-26T01:14:10.048Z [DEBUG] plugin.terraform-provider-aws_v3.18.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-aws/aws/resource_aws_rds_cluster.go:750 +0x6ea1
2020-11-26T01:14:10.048Z [DEBUG] plugin.terraform-provider-aws_v3.18.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0008cea50, 0x6e01620, 0xc000fdaf00, 0xc001e29b80, 0x55ba9a0, 0xc000fb7b80, 0x0, 0x0, 0x0)
2020-11-26T01:14:10.048Z [DEBUG] plugin.terraform-provider-aws_v3.18.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.3.0/helper/schema/resource.go:268 +0x88
2020-11-26T01:14:10.048Z [DEBUG] plugin.terraform-provider-aws_v3.18.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0008cea50, 0x6e01620, 0xc000fdaf00, 0xc0008f6770, 0xc001e66300, 0x55ba9a0, 0xc000fb7b80, 0x0, 0x0, 0x0, ...)
2020-11-26T01:14:10.062Z [DEBUG] plugin.terraform-provider-aws_v3.18.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.3.0/helper/schema/resource.go:386 +0x681
2020-11-26T01:14:10.062Z [DEBUG] plugin.terraform-provider-aws_v3.18.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000e796a0, 0x6e01620, 0xc000fdaf00, 0xc001e42910, 0xc000fdaf00, 0xc000e8ec00, 0x6e2fc60)
2020-11-26T01:14:10.062Z [DEBUG] plugin.terraform-provider-aws_v3.18.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.3.0/helper/schema/grpc_provider.go:951 +0x8b2
2020-11-26T01:14:10.062Z [DEBUG] plugin.terraform-provider-aws_v3.18.0_x5: github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc001500840, 0x6e01620, 0xc000fdaf00, 0xc0008f64d0, 0xc001500840, 0xc0012ffef0, 0xc001530ba0)
2020-11-26T01:14:10.062Z [DEBUG] plugin.terraform-provider-aws_v3.18.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.1.0/tfprotov5/server/server.go:331 +0xac
2020-11-26T01:14:10.062Z [DEBUG] plugin.terraform-provider-aws_v3.18.0_x5: github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x6049f60, 0xc001500840, 0x6e016e0, 0xc0012ffef0, 0xc000e8ec00, 0x0, 0x6e016e0, 0xc0012ffef0, 0xc001e59500, 0x9e7)
2020-11-26T01:14:10.063Z [DEBUG] plugin.terraform-provider-aws_v3.18.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.1.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x217
2020-11-26T01:14:10.063Z [DEBUG] plugin.terraform-provider-aws_v3.18.0_x5: google.golang.org/grpc.(*Server).processUnaryRPC(0xc0005fa8c0, 0x6e25ea0, 0xc0012c6180, 0xc001e25900, 0xc00094f530, 0xa5b8da0, 0x0, 0x0, 0x0)
2020-11-26T01:14:10.063Z [DEBUG] plugin.terraform-provider-aws_v3.18.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:1194 +0x50a
2020-11-26T01:14:10.063Z [DEBUG] plugin.terraform-provider-aws_v3.18.0_x5: google.golang.org/grpc.(*Server).handleStream(0xc0005fa8c0, 0x6e25ea0, 0xc0012c6180, 0xc001e25900, 0x0)
2020-11-26T01:14:10.063Z [DEBUG] plugin.terraform-provider-aws_v3.18.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:1517 +0xcfd
2020-11-26T01:14:10.063Z [DEBUG] plugin.terraform-provider-aws_v3.18.0_x5: google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0009b2170, 0xc0005fa8c0, 0x6e25ea0, 0xc0012c6180, 0xc001e25900)
2020-11-26T01:14:10.063Z [DEBUG] plugin.terraform-provider-aws_v3.18.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:859 +0xa1
2020-11-26T01:14:10.063Z [DEBUG] plugin.terraform-provider-aws_v3.18.0_x5: created by google.golang.org/grpc.(*Server).serveStreams.func1
2020-11-26T01:14:10.063Z [DEBUG] plugin.terraform-provider-aws_v3.18.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.32.0/server.go:857 +0x204
2020-11-26T01:14:10.063Z [DEBUG] plugin: plugin process exited: path=.terraform/plugins/registry.terraform.io/hashicorp/aws/3.18.0/linux_amd64/terraform-provider-aws_v3.18.0_x5 pid=1751 error="exit status 2"
2020/11/26 01:14:10 [DEBUG] aws_rds_cluster.intake_cluster: apply errored, but we're indicating that via the Error pointer rather than returning it: rpc error: code = Unavailable desc = transport is closing
2020/11/26 01:14:10 [TRACE] eval: *terraform.EvalMaybeTainted
2020/11/26 01:14:10 [TRACE] EvalMaybeTainted: aws_rds_cluster.intake_cluster encountered an error during creation, so it is now marked as tainted
2020/11/26 01:14:10 [TRACE] eval: *terraform.EvalWriteState
2020/11/26 01:14:10 [TRACE] EvalWriteState: removing state object for aws_rds_cluster.intake_cluster
2020/11/26 01:14:10 [TRACE] eval: *terraform.EvalApplyProvisioners
2020/11/26 01:14:10 [TRACE] EvalApplyProvisioners: aws_rds_cluster.intake_cluster has no state, so skipping provisioners
2020/11/26 01:14:10 [TRACE] eval: *terraform.EvalMaybeTainted
2020/11/26 01:14:10 [TRACE] EvalMaybeTainted: aws_rds_cluster.intake_cluster encountered an error during creation, so it is now marked as tainted
2020/11/26 01:14:10 [TRACE] eval: *terraform.EvalWriteState
2020/11/26 01:14:10 [TRACE] EvalWriteState: removing state object for aws_rds_cluster.intake_cluster
2020/11/26 01:14:10 [TRACE] eval: *terraform.EvalIf
2020/11/26 01:14:10 [TRACE] eval: *terraform.EvalIf
2020/11/26 01:14:10 [TRACE] eval: *terraform.EvalWriteDiff
2020/11/26 01:14:10 [TRACE] eval: *terraform.EvalApplyPost
2020/11/26 01:14:10 [ERROR] eval: *terraform.EvalApplyPost, err: rpc error: code = Unavailable desc = transport is closing
2020/11/26 01:14:10 [ERROR] eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2020/11/26 01:14:10 [TRACE] [walkApply] Exiting eval tree: aws_rds_cluster.intake_cluster
2020/11/26 01:14:10 [TRACE] vertex "aws_rds_cluster.intake_cluster": visit complete
2020/11/26 01:14:10 [TRACE] dag/walk: upstream of "aws_rds_cluster_instance.intake_instance" errored, so skipping
2020/11/26 01:14:10 [TRACE] dag/walk: upstream of "provider[\"registry.terraform.io/hashicorp/aws\"] (close)" errored, so skipping
2020-11-26T01:14:10.062Z [WARN]  plugin.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2020/11/26 01:14:10 [TRACE] dag/walk: upstream of "meta.count-boundary (EachMode fixup)" errored, so skipping
2020/11/26 01:14:10 [TRACE] dag/walk: upstream of "root" errored, so skipping
2020-11-26T01:14:10.067Z [DEBUG] plugin: plugin exited
```

### CODE REVIEW

File: aws/resource_aws_rds_cluster.go [hunk 1]
```diff
@@ -747,8 +747,8 @@ func resourceAwsRDSClusterCreate(d *schema.ResourceData, meta interface{}) error
 			createOpts.KmsKeyId = aws.String(attr.(string))
 		}
 
-		if attr, ok := d.GetOk("enabled_cloudwatch_logs_exports"); ok && len(attr.([]interface{})) > 0 {
-			createOpts.EnableCloudwatchLogsExports = expandStringList(attr.([]interface{}))
+		if attr, ok := d.GetOk("enabled_cloudwatch_logs_exports"); ok && attr.(*schema.Set).Len() > 0 {
```
Comment (anGie44) [inline aws/resource_aws_rds_cluster.go:750]: 💯

File: aws/resource_aws_rds_cluster_test.go [hunk 2]
```diff
@@ -2698,9 +2698,10 @@ resource "aws_db_subnet_group" "test" {
 }
 
 resource "aws_rds_cluster" "restored_pit" {
-  cluster_identifier  = "%s"
-  skip_final_snapshot = true
-  engine              = aws_rds_cluster.test.engine
+  cluster_identifier              = "%s"
+  skip_final_snapshot             = true
+  engine                          = aws_rds_cluster.test.engine
+  enabled_cloudwatch_logs_exports = ["audit", "error"]
```
Comment (anGie44) [inline aws/resource_aws_rds_cluster_test.go:2704]: this is a great start! let's just move this specific configuration check (with enabled cloudwatch log exports) to a new test if you don't mind so we can confirm and isolate the new expected behavior as well as check the attribute is used and set appropriately

**Review by anGie44 — CHANGES_REQUESTED** [review]
Hi @shuheiktgw, thanks for this fix! Just a request to add a new test with the updated configuration otherwise LGTM 👍

**Comment by shuheiktgw** [comment]
@anGie44 Thank you for your review! I've added a new test for the change. So would you review the PR again? 🙏 

```
 make testacc TESTARGS='-run=TestAccAWSRDSCluster_PointInTimeRestore_EnabledCloudwatchLogsExports'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSRDSCluster_PointInTimeRestore_EnabledCloudwatchLogsExports -timeout 120m
=== RUN   TestAccAWSRDSCluster_PointInTimeRestore_EnabledCloudwatchLogsExports
=== PAUSE TestAccAWSRDSCluster_PointInTimeRestore_EnabledCloudwatchLogsExports
=== CONT  TestAccAWSRDSCluster_PointInTimeRestore_EnabledCloudwatchLogsExports
--- PASS: TestAccAWSRDSCluster_PointInTimeRestore_EnabledCloudwatchLogsExports (410.03s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	411.890s
```

**Review by anGie44 — APPROVED** [review]
Looks great @shuheiktgw 🚀 

Output of acceptance tests (Commercial):
```
--- PASS: TestAccAWSRDSCluster_PointInTimeRestore (318.49s)
--- PASS: TestAccAWSRDSCluster_PointInTimeRestore_EnabledCloudwatchLogsExports (320.29s)

--- PASS: TestAccAWSRDSClusterInstance_CopyTagsToSnapshot (549.35s)
--- PASS: TestAccAWSRDSClusterInstance_namePrefix (649.35s)
--- PASS: TestAccAWSRDSClusterInstance_generatedName (688.05s)
--- PASS: TestAccAWSRDSClusterInstance_MonitoringRoleArn_RemovedToEnabled (704.28s)
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsEnabled_AuroraPostgresql (725.32s)
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsKmsKeyId_AuroraMysql1_DefaultKeyToCustomKey (759.65s)
--- PASS: TestAccAWSRDSClusterInstance_az (770.26s)
--- PASS: TestAccAWSRDSClusterInstance_kmsKey (770.38s)
--- PASS: TestAccAWSRDSClusterInstance_PubliclyAccessible (776.83s)
--- PASS: TestAccAWSRDSClusterInstance_disappears (791.37s)
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsEnabled_AuroraMysql1 (814.85s)
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsEnabled_AuroraMysql2 (816.54s)
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsKmsKeyId_AuroraMysql1 (835.74s)
--- PASS: TestAccAWSRDSCluster_basic (103.06s)
--- PASS: TestAccAWSRDSClusterInstance_isAlreadyBeingDeleted (863.98s)
--- PASS: TestAccAWSRDSCluster_AvailabilityZones (134.06s)
--- PASS: TestAccAWSRDSClusterInstance_MonitoringInterval (905.18s)
--- PASS: TestAccAWSRDSClusterInstance_MonitoringRoleArn_EnabledToRemoved (907.89s)
--- PASS: TestAccAWSRDSCluster_missingUserNameCausesError (4.16s)
--- PASS: TestAccAWSRDSCluster_BacktrackWindow (152.22s)
--- PASS: TestAccAWSRDSCluster_ClusterIdentifierPrefix (143.51s)
--- PASS: TestAccAWSRDSClusterInstance_MonitoringRoleArn_EnabledToDisabled (949.77s)
--- PASS: TestAccAWSRDSCluster_DbSubnetGroupName (137.42s)
--- PASS: TestAccAWSRDSCluster_generatedName (112.46s)
--- PASS: TestAccAWSRDSCluster_Tags (151.88s)
--- PASS: TestAccAWSRDSCluster_EnabledCloudwatchLogsExports_Postgresql (133.56s)
--- PASS: TestAccAWSRDSCluster_takeFinalSnapshot (174.59s)
--- PASS: TestAccAWSRDSCluster_updateIamRoles (146.52s)
--- PASS: TestAccAWSRDSCluster_kmsKey (134.05s)
--- PASS: TestAccAWSRDSCluster_encrypted (133.51s)
--- PASS: TestAccAWSRDSCluster_EnabledCloudwatchLogsExports_MySQL (200.73s)
--- PASS: TestAccAWSRDSCluster_copyTagsToSnapshot (221.40s)
--- PASS: TestAccAWSRDSCluster_iamAuth (122.81s)
--- PASS: TestAccAWSRDSClusterEndpoint_tags (1209.60s)
--- PASS: TestAccAWSRDSCluster_EngineMode_Multimaster (116.64s)
--- PASS: TestAccAWSRDSClusterInstance_CACertificateIdentifier (519.61s)
--- PASS: TestAccAWSRDSCluster_EngineMode_Global (163.80s)
--- PASS: TestAccAWSRDSCluster_DeletionProtection (172.10s)
--- PASS: TestAccAWSRDSCluster_backupsUpdate (202.57s)
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsKmsKeyId_AuroraPostgresql_DefaultKeyToCustomKey (564.67s)
--- PASS: TestAccAWSRDSCluster_EngineMode_ParallelQuery (162.68s)
--- PASS: TestAccAWSRDSClusterEndpoint_basic (1322.32s)
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsKmsKeyId_AuroraMysql2 (773.19s)
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsKmsKeyId_AuroraPostgresql (650.32s)
--- PASS: TestAccAWSRDSCluster_GlobalClusterIdentifier_EngineMode_Provisioned (115.75s)
--- PASS: TestAccAWSRDSCluster_GlobalClusterIdentifier_EngineMode_Global_Add (158.45s)
--- PASS: TestAccAWSRDSCluster_GlobalClusterIdentifier_EngineMode_Global (167.44s)
--- SKIP: TestAccAWSRDSCluster_SnapshotIdentifier_EngineMode_Serverless (0.00s)
--- PASS: TestAccAWSRDSCluster_GlobalClusterIdentifier_EngineMode_Global_Remove (156.67s)
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsKmsKeyId_AuroraMysql2_DefaultKeyToCustomKey (736.64s)
--- PASS: TestAccAWSRDSClusterInstance_basic (1408.97s)
--- PASS: TestAccAWSRDSCluster_GlobalClusterIdentifier_EngineMode_Global_Update (171.45s)
--- PASS: TestAccAWSRDSCluster_Port (252.54s)
--- PASS: TestAccAWSRDSCluster_EngineMode (447.30s)
--- PASS: TestAccAWSRDSCluster_EngineVersion (419.80s)
--- PASS: TestAccAWSRDSCluster_ScalingConfiguration_DefaultMinCapacity (305.65s)
--- PASS: TestAccAWSRDSCluster_SnapshotIdentifier (350.89s)
--- PASS: TestAccAWSRDSCluster_ScalingConfiguration (374.02s)
--- PASS: TestAccAWSRDSCluster_SnapshotIdentifier_EngineVersion_Equal (334.11s)
--- PASS: TestAccAWSRDSCluster_SnapshotIdentifier_KmsKeyId (353.04s)
--- PASS: TestAccAWSRDSCluster_SnapshotIdentifier_EngineMode_Provisioned (371.78s)
--- PASS: TestAccAWSRDSCluster_SnapshotIdentifier_EngineVersion_Different (374.04s)
--- PASS: TestAccAWSRDSCluster_SnapshotIdentifier_MasterUsername (342.50s)
--- PASS: TestAccAWSRDSCluster_SnapshotIdentifier_DeletionProtection (422.27s)
--- PASS: TestAccAWSRDSCluster_SnapshotIdentifier_MasterPassword (352.28s)
--- PASS: TestAccAWSRDSCluster_SnapshotIdentifier_EngineMode_ParallelQuery (420.83s)
--- PASS: TestAccAWSRDSCluster_SnapshotIdentifier_PreferredMaintenanceWindow (342.20s)
--- PASS: TestAccAWSRDSCluster_SnapshotIdentifier_PreferredBackupWindow (371.67s)
--- PASS: TestAccAWSRDSCluster_SnapshotIdentifier_Tags (382.28s)
--- PASS: TestAccAWSRDSCluster_SnapshotIdentifier_VpcSecurityGroupIds (362.95s)
--- PASS: TestAccAWSRDSCluster_SnapshotIdentifier_EncryptedRestore (371.88s)
--- PASS: TestAccAWSRDSCluster_SnapshotIdentifier_VpcSecurityGroupIds_Tags (392.90s)
--- PASS: TestAccAWSRDSCluster_EnableHttpEndpoint (397.47s)
--- PASS: TestAccAWSRDSCluster_AllowMajorVersionUpgrade (1394.58s)
--- PASS: TestAccAWSRDSCluster_EngineVersionWithPrimaryInstance (1035.79s)
--- FAIL: TestAccAWSRDSCluster_s3Restore (1521.12s) --- ongoing failure in main branch
--- PASS: TestAccAWSRDSCluster_ReplicationSourceIdentifier_KmsKeyId (1605.17s)
--- PASS: TestAccAWSRDSCluster_GlobalClusterIdentifier_PrimarySecondaryClusters (1811.47s)
--- PASS: TestAccAWSRDSCluster_GlobalClusterIdentifier_ReplicationSourceIdentifier (1821.04s)
```

**Comment by shuheiktgw** [comment]
Thank you!

**Pull request merged**